brickschema/validate.py

```diff
@@ -0,0 +1,423 @@
+    def __init__(self, useBrickSchema=True, useDefaultShapes=True):
+        self.log = logging.getLogger()
+        self.log.setLevel(logging.DEBUG if hasattr(sys, '_called_from_test') else logging.WARNING)
+        self.log.info('Validate init')
+
+        self.namespaceDict = {}
+        self.brickG = Graph()
+        self.shapeG = Graph()
+
+        if useBrickSchema:
+            data = pkgutil.get_data(__name__, "ontologies/Brick.ttl").decode()
+            self.brickG.parse(source=io.StringIO(data), format='turtle')
+            self.__buildNamespaceDict(self.brickG)
+
+            # Remove rdfs:domain and rdfs:range.  The modified
+            # ontology will be used for pySHACL reasoning.
+            # See DESIGN.md for more discussion.
+
+            self.brickG.update('DELETE { ?s rdfs:domain ?o .} WHERE { ?s rdfs:domain ?o . }',
+                               initNs=self.namespaceDict)
+            self.brickG.update('DELETE { ?s rdfs:range ?o .} WHERE { ?s rdfs:range ?o . }',
+                               initNs=self.namespaceDict)
+
+        if useDefaultShapes:
+            data = pkgutil.get_data(__name__, "ontologies/BrickShape.ttl").decode()
+            self.shapeG.parse(source=io.StringIO(data), format='turtle')
+            self.__buildNamespaceDict(self.shapeG)
+
+
+    def validate(self, data_graph, shacl_graph=None, ont_graph=None,
+                 inference='rdfs', abort_on_error=False, advanced=True,
+                 meta_shacl=True, debug=False):
+        """
+        Validates data_graph against shacl_graph and ont_graph.
+
+        Args:
+            shacl_graph: default to BrickShape.ttl and shapes added with the addShape[Graph|File]() methods
+            ont_graph: default to Brick.ttl
+
+        Returns:
+            (tuple) (conforms,  result graph, result text)
+        """
+
+        self.log.info('wrapper function for pySHACL validate()')
+
+        sg = shacl_graph if shacl_graph else self.shapeG
+        og = ont_graph if ont_graph else self.brickG
+
+        self.data_graph = data_graph
+        (self.conforms, self.results_graph, self.results_text) = pyshacl.validate(
+            data_graph, shacl_graph=sg, ont_graph=og,
+            inference=inference, abort_on_error=abort_on_error,
+            meta_shacl=meta_shacl, debug=debug)
+
+        return (self.conforms, self.results_graph, self.results_text)
+
+
+    def addShapeFile(self, shapeFile):
+        """
+        Adds additional SHACL shape file into the existing shape graph.
+        """
+        self.log.info('load shape file %s' % shapeFile)
+        g = Graph()
+        g.parse(shapeFile, format='turtle')
+        self.shapeG = self.shapeG + g
+
+
+    def addShapeGraph(self, shapeGraph):
+        """
+        Adds additional SHACL shape graph into the existing shape graph.
+        """
+        self.log.info('load shape graph')
+        self.shapeG = self.shapeG + shapeGraph
+
+
+    def accumulatedNamespaces(self):
```

Comment:
This should probably be a property:

```python
@property
def namespaces(self):
    return self.namespaceDict
```

This more closely matches the semantics of how the object is meant to be used